hwilib/devices/bitbox02.py

```diff
@@ -0,0 +1,735 @@
+
+
+class BitBoxAppNoiseCache(BitBoxNoiseConfig):
+    """
+    Noise config that reads and stores the noise keys in the same location as the BitBoxApp.
+    This way, downstream HWI users can skip implementing the pairing dialog for the BitBox02.
+    If the BitBox02 is paired to the BitBoxApp, it will work here too.
+    """
+
+    def __init__(self) -> None:
+        folder = Path(
+            os.environ.get(
+                "XDG_CONFIG_HOME",
+                # fallback
+                os.path.join(os.environ["HOME"], ".config"),
+            )
+        )
+        self._filename = folder / "bitbox" / "bitbox02" / "bitbox02.json"
+
+    def _read_cache(self) -> Any:
+        try:
+            with self._filename.open("r") as fileh:
+                return json.load(fileh)
+        except FileNotFoundError:
+            return {}
+        except json.JSONDecodeError:
+            return {}
+
+    def _write_cache(self, data: Dict[str, object]) -> None:
+        self._filename.parent.mkdir(parents=True, exist_ok=True)
+        with self._filename.open("w") as fileh:
+            json.dump(data, fileh)
+
+    def contains_device_static_pubkey(self, pubkey: bytes) -> bool:
+        data = self._read_cache()
+        return base64.b64encode(pubkey).decode() in data.get(
+            "deviceNoiseStaticPubkeys", []
+        )
+
+    def add_device_static_pubkey(self, pubkey: bytes) -> None:
+        if not self.contains_device_static_pubkey(pubkey):
+            data = self._read_cache()
+            data.setdefault("deviceNoiseStaticPubkeys", [])
+            data["deviceNoiseStaticPubkeys"].append(base64.b64encode(pubkey).decode())
+            self._write_cache(data)
+
+    def get_app_static_privkey(self) -> Optional[bytes]:
+        data = self._read_cache()
+        if "appNoiseStaticKeypair" not in data:
+            return None
+
+        return base64.b64decode(data["appNoiseStaticKeypair"]["private"])
+
+    def set_app_static_privkey(self, privkey: bytes) -> None:
+        data = self._read_cache()
+        pubkey = KeyPair25519.from_private_bytes(privkey).public_bytes
+        data["appNoiseStaticKeypair"] = {
+            "private": base64.b64encode(privkey).decode(),
+            "public": base64.b64encode(pubkey).decode(),
+        }
+        self._write_cache(data)
+
+
+class SilentNoiseConfig(BitBoxAppNoiseCache):
+    """
+    Used during `enumerate()`. Raises an exception if the device is unpaired.
+    Attestation check is silent.
+
+    Rationale: enumerate() should not show any dialogs.
+    """
+
+    def show_pairing(self, code: str, device_response: Callable[[], bool]) -> bool:
+        raise DeviceNotReadyError(_unpaired_errmsg)
+
+    def attestation_check(self, result: bool) -> None:
+        pass
+
+
+class CLINoiseConfig(BitBoxAppNoiseCache):
+    """ Noise pairing and attestation check handling in the terminal (stdin/stdout) """
+
+    def show_pairing(self, code: str, device_response: Callable[[], bool]) -> bool:
```

Comment:
It is the device response you will get by calling this (can be true or false depending on the device button clicked).

Reply:
Ah, I actually meant the method name `show_pairing` because it returns `bool` and it's unclear what it means:

```py
if x.show_pairing(...):
    # good! looks like we succeded in displaying a pairing code somewhere, probably for the user to see.
    # now what. ask user to confirm?
```

this would be more self-explanatory to me:

```py
if x.accept_pairing(...):
    # good! pairing succeded.
```

or this:

```py
if x.confirm_pairing(...):
    # all good. pairing is confirmed. can go on.
```